src/prefect/contrib/tasks/databricks/databricks.py

```diff
@@ -0,0 +1,260 @@
+        return self._timeout_seconds
+
+    @property
+    def databricks_retry_limit(self) -> int:
+        return self._databricks_retry_limit
+
+    @property
+    def databricks_retry_delay(self) -> int:
+        return self._databricks_retry_delay
+
+    @property
+    def polling_period(self) -> int:
+        return self._polling_period
+
+    @property
+    def polling_timeout(self) -> int:
+        return self._polling_timeout
+
+    @property
+    def json(self) -> dict:
+        return self._json
+
+    def _get_cluster_id(self, cluster_name: str) -> str:
+        """
+        Utility function to get a cluster id from cluster name
+        Args:
+            - name: cluster name
+
+        Returns:
+            - The cluster id
+        """
+        self.logger.info("Fetching existing clusters information ...")
+        query = f"{self._end_point}/clusters/list"
+        response = requests.get(
+            url=query,
+            timeout=self._timeout_seconds,
+            headers={
+                "Authorization": f"Bearer {self._databricks_token}",
+                "User-Agent": USER_AGENT,
+            },
+        )
+        response.raise_for_status()
+        data = response.json()
+        clusters = dict(
+            map(
+                lambda k: (k["default_tags"]["ClusterName"], k["cluster_id"]),
+                data["clusters"],
+            )
+        )
+
+        if cluster_name not in clusters:
+            raise Exception(f"Cluster {cluster_name} does not exist !")
+        return clusters[cluster_name]
+
+    def _wait_for_run_to_complete(
+        self, run_id: str, polling_period: int, timeout: int = 3600
+    ) -> dict:
+        """
+        Fetch Databricks run status via api/../jobs/run/get?run_id=<RUN_ID> until run state
+        is terminated. A run is terminated when matching either SUCCESS, FAILED, TIMEDOUT or CANCELED
+
+        Args:
+            - run_id: Databricks run Id
+            - polling_period: Delay in seconds between each API call
+            - timeout: Timeout
+
+        Returns:
+            - Run response payload as python dict,
+              see https://docs.databricks.com/dev-tools/api/latest/jobs.html#runs-get
+        """
+        query = f"{self._end_point}/jobs/runs/get?run_id={run_id}"
+        start_time = time()
+        while time() - start_time < timeout:
+            response = requests.get(
+                url=query,
+                timeout=self._timeout_seconds,
+                headers={
+                    "Authorization": f"Bearer {self._databricks_token}",
+                    "User-Agent": USER_AGENT,
+                },
+            )
+
+            response.raise_for_status()
+            data = response.json()
+            run_page, run_name, lifecycle_state = (
+                data["run_page_url"],
+                data["run_name"],
+                data["state"]["life_cycle_state"],
+            )
+
+            if "result_state" in data["state"]:
```

Comment:
I don't think checking the presence of `result_state` for task termination is correct here.   Based on https://docs.databricks.com/dev-tools/api/latest/jobs.html#runresultstate there appear to be various cases where the run can have finished but `result_state` is empty.  Instead I think you need to check whether `life_cycle_state` is a terminal state.